### ISSUE TITLE
PCL 2d qhull work properly depends on the order of points

### ISSUE THREAD

**Issue by moyushion** [PointCloudLibrary/pcl#6161]
I use the pcl-1.14.1 library to create a 2d convex hull with 4 points. But whether work properly depends on the order of points.

- **If I input the information of four points(bbox_ptr) in this order, it will be crash.**
```
    PointCloud<PointXYZ>::Ptr bbox_ptr(new PointCloud<PointXYZ>);
    bbox_ptr->push_back(PointXYZ(0,0,0));
    bbox_ptr->push_back(PointXYZ(5939,5939,0));
    bbox_ptr->push_back(PointXYZ(0,12000,0));
    bbox_ptr->push_back(PointXYZ(0,6000,0));
    ConvexHull<PointXYZ> hull;
    hull.setInputCloud(bbox_ptr);
    hull.setDimension(2);
    vector<Vertices> polygons;
    PointCloud<PointXYZ>::Ptr surface_hull (new PointCloud<PointXYZ>);
    hull.reconstruct(*surface_hull, polygons);
```
[Some error information given:]

```
QH6154 Qhull precision error: Initial simplex is flat (facet 1 is coplanar with the interior point)

While executing:  | qhull 
Options selected for Qhull 2015.2.r 2016/01/18:
  run-id 2049545305  _pre-merge  _zero-centrum  _max-width 1.2e+04
  Error-roundoff 8e-12  _one-merge 4e-11  _near-inside 2e-10
  Visible-distance 1.6e-11  U-coplanar-distance 1.6e-11  Width-outside 3.2e-11
  _wide-facet 9.7e-11

precision problems (corrected unless 'Q0' or an error)
      2 flipped facets

The input to qhull appears to be less than 2 dimensional, or a
computation has overflowed.

Qhull could not construct a clearly convex simplex from points:
- p1(v2): 5.9e+03     0
- p2(v1): 1.2e+04     0
- p0(v0):     0     0

The center point is coplanar with a facet, or a vertex is coplanar
with a neighboring facet.  The maximum round off error for
computing distances is 8e-12.  The center point, facets and distances
to the center point are as follows:

center point     5980        0

facet p2 p0 distance=    0
facet p1 p0 distance=    0
facet p1 p2 distance=    0

These points either have a maximum or minimum x-coordinate, or
they maximize the determinant for k coordinates.  Trial points
are first selected from points that maximize a coordinate.

The min and max coordinates for each dimension are:
  0:         0   1.2e+04  difference= 1.2e+04
  1:         0         0  difference=    0

If the input should be full dimensional, you have several options that
may determine an initial simplex:
  - use 'QJ'  to joggle the input and make it full dimensional
  - use 'QbB' to scale the points to the unit cube
  - use 'QR0' to randomly rotate the input for different maximum points
  - use 'Qs'  to search all points for the initial simplex
  - use 'En'  to specify a maximum roundoff error less than 8e-12.
  - trace execution with 'T3' to see the determinant for each point.

If the input is lower dimensional:
  - use 'QJ' to joggle the input and make it full dimensional
  - use 'Qbk:0Bk:0' to delete coordinate k from the input.  You should
    pick the coordinate with the least range.  The hull will have the
    correct topology.
  - determine the flat containing the points, rotate the points
    into a coordinate plane, and delete the other coordinates.
  - add one or more points to make the input full dimensional.
[pcl::ConvexHull::performReconstrution2D] ERROR: qhull was unable to compute a convex hull for the given point cloud (4)!
```

- **Buf if I change the order of four points(bbox_ptr), it will be work.**
```
    PointCloud<PointXYZ>::Ptr bbox_ptr(new PointCloud<PointXYZ>);    
    bbox_ptr->push_back(PointXYZ(5939,5939,0));
    bbox_ptr->push_back(PointXYZ(0,12000,0));
    bbox_ptr->push_back(PointXYZ(0,6000,0));
    bbox_ptr->push_back(PointXYZ(0,0,0));
    ConvexHull<PointXYZ> hull;
    hull.setInputCloud(bbox_ptr);
    hull.setDimension(2);
    vector<Vertices> polygons;
    PointCloud<PointXYZ>::Ptr surface_hull (new PointCloud<PointXYZ>);
    hull.reconstruct(*surface_hull, polygons);
```

I am searching for a long time on net. But no use. Some help will be appreciated :)

**Comment by mvieth** [comment]
Hi, I believe this problem is fixed on the master branch (by pull request https://github.com/PointCloudLibrary/pcl/pull/6129 ). Can you build from source and try again?

**Comment by moyushion** [comment]
> Hi, I believe this problem is fixed on the master branch (by pull request #6129 ). Can you build from source and try again?

Yes, the master branch do work well. This problem no longer exists. Thanks a lot!

**Comment by mvieth** [comment]
Great!
